core/src/rpc_subscriptions.rs

```diff
@@ -576,15 +609,17 @@ pub(crate) mod tests {
 
     pub(crate) fn robust_poll_or_panic<T: Debug + Send + 'static>(
         receiver: futures::sync::mpsc::Receiver<T>,
-    ) -> T {
+    ) -> (T, futures::sync::mpsc::Receiver<T>) {
         let (inner_sender, inner_receiver) = channel();
         let mut rt = Runtime::new().unwrap();
         rt.spawn(futures::lazy(|| {
             let recv_timeout = receiver
                 .into_future()
                 .timeout(Duration::from_millis(RECEIVE_DELAY_MILLIS))
                 .map(move |result| match result {
-                    (Some(value), _) => inner_sender.send(value).expect("send error"),
+                    (Some(value), receiver) => {
+                        inner_sender.send((value, receiver)).expect("send error")
```

Comment:
❗️ This is way easier than what I was trying to do with this function last week. Awesome!

Reply:
Yes! Such a relief when I realized this was possible haha